### PR TITLE
Feature/1411 make dynamic p4 settings

### PR DESCRIPTION
**Pull request by sagarsdeshmukh**
Update static references of ACT, EXPLORE and ISSUE CATEGORY page by Planet4 admin settings

### CODE REVIEW

File: classes/controller/blocks/class-p4bks-blocks-campaignthumbnail-controller.php [hunk 1]
```diff
@@ -59,18 +59,26 @@ public function prepare_template( $fields, $content, $shortcode_tag ) : string {
...
+				$context_tags     = get_queried_object();
+
+				$parent_id = planet4_get_option( 'select_explore_page' );
+				if( 0 != $parent_id ) {
+					$args = array(
+						'post_type'      => 'page',
+						'post_parent'    => $parent_id
+					);
+				} else {
+					$page              = get_page_by_path( 'explore' );
```
Comment (koyan) [inline classes/controller/blocks/class-p4bks-blocks-campaignthumbnail-controller.php]: Why do you leave this code here? 
Please remove the old code from all the parts

Comment (sagarsdeshmukh) [reply classes/controller/blocks/class-p4bks-blocks-campaignthumbnail-controller.php]: @koyan - If admin planet4 settings are missing, in that case, we need failover condition handling, for that purpose I added a condition here.

Comment (koyan) [reply classes/controller/blocks/class-p4bks-blocks-campaignthumbnail-controller.php]: no, gracefully do not show anything. Or throw a nice explanation to the user. But do not keep in there a huge block of code that offers nothing.

File: classes/controller/blocks/class-takeactionboxout-controllers.php [hunk 2]
```diff
@@ -21,10 +21,15 @@ class TakeActionBoxout_Controller extends Controller {
 		public function prepare_fields() {
 
 			// Get the id of the ACT page. We need this to get the children posts/pages of the ACT Page.
-			$arguments = [
-				'post_type'     => 'page',
-				'post_name__in' => [ 'act', 'ACT', 'Act' ],
-			];
+			$parent_act_id = planet4_get_option( 'select_act_page' );
```
Comment (koyan) [inline classes/controller/blocks/class-takeactionboxout-controllers.php]: Please change this variable from "select_act_page" to "act_page" (everywhere where it is relevant).  Names must be logical. It is not logical to have a name with the "select_" in front.

File: classes/controller/blocks/class-splittwocolumns-controller.php [hunk 3]
```diff
@@ -20,11 +20,15 @@ class SplitTwoColumns_Controller extends Controller {
 		 */
 		public function prepare_fields() {
 
-			$categories = get_categories( [
-				'parent'  => get_cat_ID( 'Issues' ),            // Issue categories needs to be children of category Issues.
-				'orderby' => 'name',
-				'order'   => 'ASC',
-			] );
+			$issue_category_id = planet4_get_option( 'select_category' );
```
Comment (koyan) [inline classes/controller/blocks/class-splittwocolumns-controller.php]: Please change the name of the parameter from "select_category" to "issues_parent_category" in all places relevant.

File: classes/controller/blocks/class-covers-controllers.php [hunk 4]
```diff
@@ -87,22 +87,25 @@ public function prepare_fields() {
...
-				'orderby'     => 'post_date',
-				'order'       => 'DESC',
-				'numberposts' => P4BKS_COVERS_NUM,
-			];
-
-			// If user selected a tag to associate with the Take Action page covers.
-			if ( $select_tags ) {
-				$tag_ids = explode( ',', $select_tags );
-				$args['tag__in'] = $tag_ids;
+			$parent_act_id  = planet4_get_option( 'select_act_page' );
```
Comment (koyan) [inline classes/controller/blocks/class-covers-controllers.php]: Please change (in all places relevant) the name of the pareameter from "select_act_page", to "act_page"

File: classes/controller/blocks/class-campaignthumbnail-controller.php [hunk 5]
```diff
@@ -57,17 +57,18 @@ public function prepare_template( $fields, $content, $shortcode_tag ) : string {
...
-				$context_tags  = get_queried_object();
-
-				$args = array(
-					'post_type'      => 'page',
-					'post_parent'    => $parent_id,
-				);
-				$explore_children = get_children( $args );
+				$category         = get_category( $fields['category_id'] );
+				$context_tags     = get_queried_object();
+				$parent_id        = planet4_get_option( 'select_explore_page' );
```
Comment (koyan) [inline classes/controller/blocks/class-campaignthumbnail-controller.php]: Change param name

**Comment by koyan** [comment]
Looks good 👍

**Pull request merged**